src/time/src/lib.rs

```diff
@@ -99,7 +98,19 @@ pub extern "C" fn asctime_r(tm: *const tm, buf: *mut c_char) -> *mut c_char {
 
 #[no_mangle]
 pub extern "C" fn clock() -> clock_t {
-    unimplemented!();
+    let mut ts: timespec = unsafe { core::mem::uninitialized() };
+
+    if clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &mut ts) != 0 {
+        return -1;
+    }
+
+    if ts.tv_sec > time_t::max_value() / CLOCKS_PER_SEC
+        || ts.tv_nsec / 1000 > time_t::max_value() - CLOCKS_PER_SEC * ts.tv_sec
```

Comment:
I also think the `ts.tv_nsec / 1000` should be implemented in terms of `CLOCKS_PER_SEC`.  It should be something like the [FreeBSD version](https://github.com/lattera/freebsd/blob/master/lib/libc/gen/clock.c), but using nanoseconds instead of microseconds.

Reply:
Eh, I think this version is a bit simpler, to be honest. At some point the extra mental overhead from implementing it in terms of something else isn't worth it for simple conditions like this.

Reply:
If the definition of `CLOCKS_PER_SEC` were to change (which is admittedly unlikely unless we stopped trying to be POSIX-compliant), the function would return an incorrect result.  It's also just `ts.tv_nsec / (1_000_000_000 / CLOCKS_PER_SEC)`, so honestly I don't think it's that complicated.